controllers/machine.go

```diff
@@ -468,6 +473,102 @@ func byHardwareAffinity(hardware []tinkv1.Hardware, preferred []infrastructurev1
 	}, nil
 }
 
+// ensureHardwareReadyToProvision ensures the hardware is ready to be provisioned.
+// Uses the BMCRef from the hardware to create a BMCJob.
+// The BMCJob is responsible for getting the machine to desired state for provisioning.
+func (mrc *machineReconcileContext) ensureHardwareReadyToProvision(hardware *tinkv1.Hardware) error {
+	if hardware.Spec.BMCRef == nil {
+		mrc.log.Info("Skipping BMC state management", "BMCRef", hardware.Spec.BMCRef, "Hardware", hardware.Name)
+
+		return nil
+	}
+
+	jobName := fmt.Sprintf("%s-provision", mrc.tinkerbellMachine.Name)
+	// Check if BMC Job already exists for the machine
+	exists, err := mrc.bmcJobExists(jobName)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		return nil
+	}
```

Comment:
Do you mean a use case where someone manually created a job object? 

because since we have ownerRefs we would not have a case where the machine is deleted and the job is still around.